src/ruby/travelling-salesman.rb

```diff
@@ -27,81 +26,55 @@ def shortestPath
 
     # add option to current order and distance
     private
-    def addOption(distance, node)
-        nodeIndex = @nodes.index(node)
-        for row in @matrix 
-            row[nodeIndex] = 0
-        end
+    def addOption(distance, index)
+        @matrix.each {|row| row[index] = 0}
 
-        @order << node
+        @order << @nodes[index]
         @distance += distance
     end
 
     # goes to next node to add to route
-    private
-    def addNextNode(distance, node)
-        nodeIndex = @nodes.index(node)
-
-        if (noOptionsAvailable(@matrix[nodeIndex]))
+    def addNextNode(distance, index)
+        if (noOptionsAvailable(@matrix[index]))
             updateShortest
-            goBackNode(distance, node)
+            goBackNode(distance, index)
         else
-            for option in @matrix[nodeIndex]
-                nextNodeIndex = @matrix[nodeIndex].index(option)
-                nextNode = @nodes[nextNodeIndex]
-                if (option != 0)
-                    addOption(option, nextNode)
-                    addNextNode(option, nextNode)
-                end
+            @matrix[index].each_with_index do |option, index|
+              if (option != 0)
+                  addOption(option, index)
+                  addNextNode(option, index)
+              end
             end
-
-            goBackNode(distance, node)
+            goBackNode(distance, index)
         end
     end
 
     # check to see if at a dead end
-    private
     def noOptionsAvailable(row)
-        deadEnd = true
-        for cell in row
-            if( cell != 0)
-                deadEnd = false
-            end
-        end
-
-        return deadEnd
+        row.none? {|cell| cell != 0}
```

Comment:
Methods in Ruby which end with a question mark mean they will return a boolean. There are loads of methods in the `Enumerable` module to help you with traversal over objects/arrays. Here is a small selection of ones you can use `all?, any? find, find_all, include? max, min, none?, one?`

Arrays extends the Enumerable module so it has access to all of them 
https://ruby-doc.org/core-2.5.0/Enumerable.html